provider/kubernetes/kubernetes.go

```diff
@@ -375,23 +380,38 @@ func (p *Provider) loadConfig(templateObjects types.Configuration) *types.Config
 	return configuration
 }
 
-func getRuleForPath(pa v1beta1.HTTPIngressPath, i *v1beta1.Ingress) string {
+func getRuleForPath(pa v1beta1.HTTPIngressPath, i *v1beta1.Ingress) (string, error) {
 	if len(pa.Path) == 0 {
-		return ""
+		return "", nil
 	}
 
 	ruleType := i.Annotations[label.TraefikFrontendRuleType]
 	if ruleType == "" {
 		ruleType = ruleTypePathPrefix
 	}
 
+	var pathReplaceAnnotation string
+	if ruleType == ruleTypeReplacePath {
+		pathReplaceAnnotation = label.TraefikFrontendRuleType
+	}
+
 	rules := []string{ruleType + ":" + pa.Path}
 
 	if rewriteTarget := i.Annotations[annotationKubernetesRewriteTarget]; rewriteTarget != "" {
+		if pathReplaceAnnotation != "" {
+			return "", fmt.Errorf("rewrite-target must not be used together with annotation %q", pathReplaceAnnotation)
+		}
 		rules = append(rules, ruleTypeReplacePath+":"+rewriteTarget)
+		pathReplaceAnnotation = annotationKubernetesRewriteTarget
 	}
 
-	return strings.Join(rules, ";")
+	if rootPath := label.GetStringValue(i.Annotations, annotationKubernetesAppRoot, ""); rootPath != "" && pa.Path == "/" {
+		if pathReplaceAnnotation != "" {
+			return "", fmt.Errorf("app-root must not be used together with annotation: %q", pathReplaceAnnotation)
```

Comment:
Nit-pick level 11: Please remove the colon after _annotation_ to be consistent with the other error message. :-)